src/allmydata/grid_manager.py

```diff
@@ -0,0 +1,475 @@
+"""
+Functions and classes relating to the Grid Manager internal state
+"""
+
+import sys
+from datetime import (
+    datetime,
+)
+from typing import Optional, Union
+
+from twisted.python.filepath import FilePath
+
+from allmydata.crypto import (
+    ed25519,
+)
+from allmydata.util import (
+    base32,
+    jsonbytes as json,
+    dictutil,
+)
+
+from attrs import define, asdict, Factory
+
+
+@define
+class SignedCertificate(object):
+    """
+    A signed certificate.
+    """
+    # A JSON-encoded, UTF-8-encoded certificate.
+    certificate : bytes
+
+    # The signature in base32.
+    signature : bytes
```

Comment:
This is incompatible with the suggestion in https://github.com/tahoe-lafs/tahoe-lafs/pull/1206#discussion_r990447390 -- because it needs to be a "json-able" object to use `attr.asdict` directly, so I'll take this suggestion over the other one.

Reply:
(but maybe `.marshal()` is a more-consistent name..)